### PR TITLE
docs(vue-component-meta): updated README

### PR DESCRIPTION
**Pull request by jd-solanki**
As per our discord [chat](https://discord.com/channels/793943652350427136/1027819645677350912) I have updated `vue-component-meta` readme according to the bast of my knowledge.

I hope you like it.

### CODE REVIEW

File: vue-language-tools/vue-component-meta/README.md [hunk 1]
```diff
@@ -1,10 +1,166 @@
...
+
+const checkerOptions: MetaCheckerOptions = {
+  forceUseTs: true,
+  schema: { ignore: ['MyIgnoredNestedProps'] },
+  printer: { newLine: 1 },
+}
+
+const tsconfigChecker = createComponentMetaChecker(
+  // Write your tsconfig path
+  resolve(__dirname, 'path-to-tsconfig'),
```
Comment (rchl) [inline vue-language-tools/vue-component-meta/README.md:21]: Maybe also import `resolve` to make the example complete.

File: vue-language-tools/vue-component-meta/README.md [hunk 2]
```diff
@@ -1,10 +1,166 @@
...
+
+You can use it to document your component as you build your project without writing additional documentation.
+
+## Pitfalls 👀
+
+As `vue-component-meta` uses static code analysis, it can't extract the dynamic prop definition.
+
+### default value
+
+`vue-component-meta` won't able to extract default value for prop as props can't be analyzed.
```
Comment (rchl) [inline vue-language-tools/vue-component-meta/README.md:63]:
```suggestion
`vue-component-meta` won't be able to extract default value for prop as props can't be analyzed.
```

File: vue-language-tools/vue-component-meta/README.md [hunk 3]
```diff
@@ -1,10 +1,166 @@
...
+    variant: {
+      default: 'light',
+    },
+  }),
+}
+```
+
+### description
+
+Same as above scenario you might have issue with description not generating when prop definition is dynamic. In this case write prop description can be tricky.
```
Comment (rchl) [inline vue-language-tools/vue-component-meta/README.md:101]:
```suggestion
Same as above scenario you might have issue with description not generating when prop definition is dynamic. In this case writing prop description can be tricky.
```

**Review by rchl — COMMENTED** [review]
Very nice!

**Comment by rchl** [comment]
I haven't had a chance to use `vue-component-meta` yet but I'm assuming it's able to handle cases where prop is of `Object` type and more specific type is provided through JSDoc's `@type`? Would it be worth including an example for that?

**Comment by jd-solanki** [comment]
It will be great for completion, but first, let me check if it's working fine in Anu as I don't have any playground to test it 😉 

Will be back soon.

Thanks for your suggestions 👍🏻

**Comment by rchl** [comment]
Just an addendum to that... I meant an example using `@type` as that is what one would use in JS code. If inside TS code then it would be done with `as Foo` instead, I presume. You seem to be already assuming TS as you have one example using `as`. It might be beneficial to include examples for both TS and JS.

**Comment by jd-solanki** [comment]
> Just an addendum to that... I meant an example using `@type` as that is what one would use in JS code. If inside TS code then it would be done with `as Foo` instead, I presume. You seem to be already assuming TS as you have one example using `as`. It might be beneficial to include examples for both TS and JS.

You mean like this:
```ts
props: {
  /*
   *
   * Array of texts to render as a message
   * @type {string[]}
  */
  texts: {
    type: Array,
    required: false,
  }
}
```

Is this enough?

Sorry if I didn't get your idea.

**Comment by rchl** [comment]
> You mean like this:

Yes, something like this although I believe the correct type to use would be be:

```js
props: {
  /**
   * Array of texts to render as a message
   * @type {import('vue').PropOptions<string[]>}
   */
  texts: {
    type: Array,
    required: false,
  }
}
```

**Comment by jd-solanki** [comment]
Thanks! Update the md.

File: vue-language-tools/vue-component-meta/README.md [hunk 4]
```diff
@@ -1,10 +1,195 @@
...
+### Defining type in JavaScript projects
+
+You can use JSDoc's `@type` to define the prop type in JavaScript projects.
+
+```ts
+props: {
+  /*
+   * Array of texts to render as a message
+   * @type {import('vue').PropOptions<string[]>}
+  */
```
Comment (rchl) [inline vue-language-tools/vue-component-meta/README.md:76]: Sorry to be a pain but proper JSDoc comment would be formatted like:

```suggestion
  /**
   * Array of texts to render as a message
   * @type {import('vue').PropOptions<string[]>}
   */
```

Comment (rchl) [reply vue-language-tools/vue-component-meta/README.md:76]: Also I wonder if generated types are correct in this case....

Comment (jd-solanki) [reply vue-language-tools/vue-component-meta/README.md:76]: I haven't tried it as I am on TS projects always.

Do you want me to check on JS, Maybe I can do that tomorrow we will be leaving office soon.

Sorry for that JSDoc style.

Comment (rchl) [reply vue-language-tools/vue-component-meta/README.md]: I'm sorry but this doesn't seem to work. I've tried it on tests in this repo and for prop defined this way:

```
		/**
		 * Default function Array
		 * @type {import('vue').Prop<[1, 2, 3]>}
		 */
		arrayDefault: {
			type: Array,
			default: () => [1, 2, 3]
		},
```

gotten results like:

```
    {
      "name": "arrayDefault",
      "global": false,
      "description": "Default function Array",
      "tags": [
        {
          "name": "type",
          "text": "{import('vue').Prop<number[]>}"
        }
      ],
      "required": false,
      "type": "unknown[] | undefined",
      "schema": {
        "kind": "enum",
        "type": "unknown[] | undefined",
        "schema": [
          "undefined",
          {
            "kind": "array",
            "type": "unknown[]",
            "schema": [
              "unknown"
            ]
          }
        ]
      },
      "default": "[1, 2, 3]"
    },
```

So the resulting type is `unknown[]` which means the JSDoc annotation is not taken into account.

(I've used `Prop` type here as `PropOptions` doesn't seem to be exported in Vue 3.x anymore.)

So it's best to just remove this case from the readme. Sorry for the noise.

**Comment by jd-solanki** [comment]
@rchl Done 😇

**Comment by johnsoncodehk** [comment]
Great work, thank you!

**Pull request merged**